src/presentation/http/router/note.test.ts

```diff
@@ -518,6 +564,221 @@ describe('Note API', () => {
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: childNote.id,
+        isPublic: true,
+      });
+
+      /** Create note relation between parent and grandParentNote */
+      await global.db.insertNoteRelation({
+        parentId: grandParentNote.id,
+        noteId: parentNote.id,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: parentNote.id,
+        noteId: childNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: grandParentNote.publicId,
+            content: grandParentNote.content,
+          },
+          {
+            id: parentNote.publicId,
+            content: parentNote.content,
+          },
+          {
+            id: childNote.publicId,
+            content: childNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns two parents note in case where the note is not created by the same user but share relation with status 200', async () => {
+      /** Create test user */
+      const user = await global.db.insertUser();
+
+      /** Create another user */
+      const anotherUser = await global.db.insertUser();
+
+      /** Create acces token for the user */
+      const accessToken = global.auth(user.id);
+
+      /** Create test note - a parent note */
+      const parentNote = await global.db.insertNote({
+        creatorId: anotherUser.id,
+      });
+
+      /** Create test note - a child note */
+      const childNote = await global.db.insertNote({
+        creatorId: user.id,
+      });
+
+      /** Create test note settings */
+      await global.db.insertNoteSetting({
+        noteId: childNote.id,
+        isPublic: true,
+      });
+
+      /** Create test note relation */
+      await global.db.insertNoteRelation({
+        parentId: parentNote.id,
+        noteId: childNote.id,
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${childNote.publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      expect(response?.json()).toMatchObject({
+        parents: [
+          {
+            id: parentNote.publicId,
+            content: parentNote.content,
+          },
+          {
+            id: childNote.publicId,
+            content: childNote.content,
+          },
+        ],
+      });
+    });
+
+    test('Returns one note in case where there is no relation exist for the note with status 200', async () => {
```

Comment:
case is also unclear for me. If note has no relation, why do you return a some note?